identifyyeastcells.py

```diff
@@ -0,0 +1,1146 @@
+
+    from cellprofiler.modules import identify as cpmi
+    import cellprofiler.module as cpm
+    import cellprofiler.image as cpi
+    import cellprofiler.measurement as cpmeas
+    import cellprofiler.setting as cps
+    import cellprofiler.preferences as cpp
+    from cellprofiler.gui.help import HELP_ON_MEASURING_DISTANCES
+    import cellprofiler.preferences as pref
+    import centrosome.outline
+    from centrosome.filter import laplacian_of_gaussian
+
+#################################
+#
+# Specific imports
+#
+##################################
+
+    from cellstar.utils.params_util import default_parameters, create_size_weights
+    from cellstar.segmentation import Segmentation
+    from cellstar.parameter_fitting.pf_runner import run_pf, run_rank_pf
+    from cellstar.utils.debug_util import memory_profile, speed_profile, explorer_expected
+
+except ImportError as e: 
+    # in new version 2.12 all the errors are properly shown in console (Windows)
+    home = expanduser("~") # in principle it is system independent
+    with open(pj(home,"cs_log.txt"), "a+") as log:
+        log.write("Import exception")
+        log.write(e.message)
+    raise
+
+
+###################################
+#
+# Constants
+#
+###################################
+
+DEBUG = 1
+F_BACKGROUND = "Background"
+BKG_CURRENT = 'Actual image'
+BKG_FIRST = 'First image'
+BKG_FILE = 'File'
+
+C_OBJECT_FEATURES = "Features"
+
+FTR_OBJECT_QUALITY = "Quality"
+'''The object quality - floating number the higher the better'''
+M_OBJECT_FEATURES_OBJECT_QUALITY= '%s_%s' % (C_OBJECT_FEATURES, FTR_OBJECT_QUALITY)
+
+def hack_add_from_file_into_EditObjects(dialog_box):
+    import wx
+    def on_load(event):
+        with wx.FileDialog(None,
+           message="Select image with labels",
+           wildcard="Image file (*.tif,*.tiff,*.jpg,*.jpeg,*.png,*.gif,*.bmp)|*.tif;*.tiff;*.jpg;*.jpeg;*.png;*.gif;*.bmp|*.* (all files)|*.*",
+           style=wx.FD_OPEN) as dlg:
+            if dlg.ShowModal() != wx.ID_OK:
+                return
+            path = dlg.Path
+
+        labels_loaded = (bioformats.load_image(path) * 255).astype(int)
+        if labels_loaded.ndim == 3:
+            labels_loaded = np.sum(labels_loaded, 2) / labels_loaded.shape[2]
+
+        for l in np.unique(labels_loaded):
+            if l != 0:
+                dialog_box.add_label(l == labels_loaded)
+        # no shape check, no float check
+        dialog_box.init_labels()
+        dialog_box.display()
+        dialog_box.record_undo()
+
+    ID_ACTION_LOAD_FROM_FILE = wx.NewId()
+    sizer = dialog_box.toolbar.ContainingSizer
+    load_file_button = wx.Button(dialog_box, ID_ACTION_LOAD_FROM_FILE, "Add from file")
+    list(sizer.Children)[-1].Sizer.Add(load_file_button,0, wx.ALIGN_CENTER)
+    dialog_box.Bind(wx.EVT_BUTTON, on_load, load_file_button)
+
+###################################
+#
+# The module class
+#
+###################################
+
+
+class IdentifyYeastCells(cpmi.Identify):
```

Comment:
We've recently added a new class `cellprofiler.module.ImageSegmentation`, which we're using in favor of `cpmi.Identify`. If you inherit from `cellprofiler.module.ImageSegmentation` you can call a single function [add_measurements](https://github.com/CellProfiler/CellProfiler/blob/master/cellprofiler/module.py#L976-L1022) to add image and object measurements. Additionally, the remaining [measurement methods](https://github.com/CellProfiler/CellProfiler/blob/master/cellprofiler/module.py#L1073-L1132) are automatically defined, and can be accessed by calling `super`.

To use the new module class, you'll need to modify your input image and output objects definitions in the following way:

```python
def create_settings(self):
    super(IdentifyYeastCells, self).create_settings()

    # self.input_image_name --> self.x_name
    self.x_name.doc = "How do you call the images you want to use to identify objects?"

    # self.object_name --> self.y_name
    self.y_name.doc = "How do you want to call the objects identified by this module?"
```